### PR TITLE
[BACKLOG-31040] Add hadoop clusters to open/save

### PR DESCRIPTION
**Pull request by bmorrise**

### CODE REVIEW

**Comment by buildguy** [comment]
### :gear: Build finished in 1m 32s with command:
```
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd -Daudit -P !assemblies -pl assemblies/features,impl,impl/browse
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|411|0|0|

|[Details](http://wingman2.pentaho.net/job/projects/job/big-data-plugin/job/PR-1791/1/testReport)|
|----|

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 1m 34s with command:
```
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd -Daudit -P !assemblies -pl assemblies/features,impl,impl/browse
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|411|0|0|

|[Details](http://wingman2.pentaho.net/job/projects/job/big-data-plugin/job/PR-1791/2/testReport)|
|----|

:information_source:  _This is an automatic message_

**Comment by bmorrise** [comment]
@ddiroma

**Comment by buildguy** [comment]
### :gear: Build finished in 1m 37s with command:
```
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd -Daudit -P !assemblies -pl assemblies/features,impl,impl/browse
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|411|0|0|

|[Details](http://wingman2.pentaho.net/job/projects/job/big-data-plugin/job/PR-1791/3/testReport)|
|----|

:information_source:  _This is an automatic message_

File: impl/browse/pom.xml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    </dependency>
+    <dependency>
+      <groupId>org.pentaho</groupId>
+      <artifactId>shim-api</artifactId>
+      <version>${pentaho-hadoop-shims.version}</version>
+    </dependency>
+  </dependencies>
+
+  <build>
+    <resources>
```
Comment (graimundo) [inline impl/browse/pom.xml]: @pentaho/r2d2  @bmorrise Please do not explicitly declare a `<resources>` element on a pom. Unfortunately this completely overwrites the element which prevents us from incrementally adding resources when needed (such as conventioned resources on the parent POM). Use the `build-helper-maven-plugin` instead to ensure that resources declared from above are not overwritten. 
You can find examples here: https://github.com/search?p=1&q=org%3Apentaho+%22add-resource%22&type=Code

File: impl/browse/pom.xml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+          <include>**/*</include>
+        </includes>
+        <excludes>
+          <exclude>META-INF/**/*</exclude>
+          <exclude>OSGI-INF/**/*</exclude>
+          <exclude>app/index.html</exclude>
+        </excludes>
+      </resource>
+      <resource>
+        <filtering>true</filtering>
```
Comment (graimundo) [inline impl/browse/pom.xml]: @pentaho/r2d2  @bmorrise a better alternative for filtered resources is to use the convention `resources-filtered` folder. 
https://github.com/pentaho/maven-parent-poms/blob/master/pom.xml#L1180
By using the convention it seems to me that you would just need to put your resources and filtered resources (META-INF, OSGI-INF and app/index.html) in the right folder and the pom would be simpler.

File: impl/browse/pom.xml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        <directory>src/main/resources</directory>
+        <includes>
+          <include>META-INF/**/*</include>
+          <include>OSGI-INF/**/*</include>
+          <include>app/index.html</include>
+        </includes>
+      </resource>
+      <resource>
+        <directory>src/main/javascript</directory>
```
Comment (graimundo) [inline impl/browse/pom.xml]: @pentaho/r2d2 @bmorrise For OSGi bundles, check out the javascript profiles on the parent pom that already does compilation, transpilation, adding source maps on snapshots, copying the code to the bundle and other stuff. Using the conventions from the parent pom will save you work and simply the overall management. You can always overwrite or change things on a descendant pom but for 98% of the cases what the parent pom has should suffice. 
https://github.com/pentaho/maven-parent-poms/blob/master/pentaho-ce-parent-pom/pentaho-ce-jar-parent-pom/pentaho-ce-bundle-parent-pom/pom.xml

**Pull request merged**